inference/core/interfaces/http/http_api.py

```diff
@@ -1310,6 +1329,79 @@ async def sam_segment_image(
                         )
                     return model_response
 
+            if CORE_MODEL_SAM2_ENABLED:
+
+                @app.post(
+                    "/sam2/embed_image",
+                    response_model=Sam2EmbeddingResponse,
+                    summary="SAM2 Image Embeddings",
+                    description="Run the Meta AI Segmant Anything 2 Model to embed image data.",
+                )
+                @with_route_exceptions
+                async def sam2_embed_image(
+                    inference_request: Sam2EmbeddingRequest,
+                    request: Request,
+                    api_key: Optional[str] = Query(
+                        None,
+                        description="Roboflow API Key that will be passed to the model during initialization for artifact retrieval",
+                    ),
+                ):
+                    """
+                    Embeds image data using the Meta AI Segmant Anything Model (SAM).
+
+                    Args:
+                        inference_request (SamEmbeddingRequest): The request containing the image to be embedded.
+                        api_key (Optional[str], default None): Roboflow API Key passed to the model during initialization for artifact retrieval.
+                        request (Request, default Body()): The HTTP request.
+
+                    Returns:
+                        M.Sam2EmbeddingResponse or Response: The response affirming the image has been embedded
+                    """
+                    logger.debug(f"Reached /sam2/embed_image")
+                    sam2_model_id = load_sam2_model(inference_request, api_key=api_key)
+                    model_response = await self.model_manager.infer_from_request(
+                        sam2_model_id, inference_request
+                    )
+                    return model_response
+
+                @app.post(
+                    "/sam2/segment_image",
+                    response_model=Sam2SegmentationResponse,
+                    summary="SAM2 Image Segmentation",
+                    description="Run the Meta AI Segmant Anything 2 Model to generate segmenations for image data.",
+                )
+                @with_route_exceptions
+                async def sam2_segment_image(
+                    inference_request: Sam2SegmentationRequest,
+                    request: Request,
```

Comment:
Am I understanding correctly that this doesn't accept boxes or positive/negative points? Only returns all the segmentations for the full image?

Reply:
No, if you look at the [Sam2SegmentationRequest object](https://github.com/roboflow/inference/blob/775094581ef63ec4e2166d4ed6427e93079667aa/inference/core/entities/requests/sam2.py#L59), you will see that it accepts positive/negative points

Reply:
Not accepting boxes right now though